src/main/java/com/nvd/footballmanager/service/MemberService.java

```diff
@@ -1,4 +1,127 @@
+import com.nvd.footballmanager.exceptions.AccessDeniedException;
+import com.nvd.footballmanager.mappers.MemberMapper;
+import com.nvd.footballmanager.mappers.TeamMapper;
+import com.nvd.footballmanager.model.entity.Member;
+import com.nvd.footballmanager.model.entity.MembershipRequest;
+import com.nvd.footballmanager.model.entity.Team;
+import com.nvd.footballmanager.model.entity.User;
+import com.nvd.footballmanager.model.enums.MemberRole;
+import com.nvd.footballmanager.repository.MemberRepository;
+import com.nvd.footballmanager.repository.MembershipRequestRepository;
+import com.nvd.footballmanager.repository.TeamRepository;
+import com.nvd.footballmanager.repository.UserRepository;
+import com.nvd.footballmanager.utils.Constants;
+import jakarta.persistence.EntityNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
 public class MemberService {
+
+    private final MemberRepository memberRepository;
+    private final TeamRepository teamRepository;
+    private final UserRepository userRepository;
+    private final UserService userService;
+    private final MembershipRequestRepository requestRepository;
+
+    private final MemberMapper memberMapper;
+    private final TeamMapper teamMapper;
+
+
+    public boolean isNotManagerPermission(UUID teamId) {
+        Optional<Member> member = memberRepository.findByUserIdAndTeamId(userService.getCurrentUser().getId(), teamId);
+        return !(member.isPresent() && member.get().getRole().equals(MemberRole.MANAGER));
+    }
+
+
+    @Transactional
+    public MemberDTO addMember(UUID teamId, UUID userId, MemberDTO memberDTO) {
+        Team team = teamRepository.findById(teamId)
+                .orElseThrow(EntityNotFoundException::new);
+
+        User user = userRepository.findById(userId)
+                .orElseThrow(EntityNotFoundException::new);
+
+        // chekc if user is already a member of team
+        Optional<Member> existingMember = memberRepository.findByUserIdAndTeamId(userId, teamId);
+        if (existingMember.isPresent()) {
+            throw new IllegalArgumentException(Constants.ALLREADY_MEMBER);
+        }
+
+        Member member = memberMapper.convertToEntity(memberDTO);
+        member.setTeam(team);
+        member.setUser(user);
+
+        Member savedMember = memberRepository.save(member);
+
+        return memberMapper.convertToDTO(savedMember);
+    }
+
+    public MemberDTO getMemberInfo(UUID teamId, UUID memberId) {
+        teamRepository.findById(teamId)
+                .orElseThrow(EntityNotFoundException::new);
+
+        Member member = memberRepository.findByIdAndTeamId(memberId, teamId)
+                .orElseThrow(EntityNotFoundException::new);
+
+        return memberMapper.convertToDTO(member);
+    }
+
```

Comment:
Why this method dont have @Transactional ?, the addMember and updateMemberInfo both is annotated with @Transactional

Reply:
dạ cái @Transactional em đặt lên class Service luôn được không anh ?
@Transactional
public class MemberService

- hay là chỉ nên đặt trên các method post, put và delete ạ.
- Với lại thì các api của phần này em test thì cũng đã run hết rồi ạ.
https://docs.google.com/spreadsheets/d/1gs43Oa_nfqvlIu6NZQL84_MhvmkqMHPD9ahraDg5vow/edit?gid=0#gid=0